lectures/tools_and_techniques/ar1_processes.md

```diff
@@ -0,0 +1,575 @@
+* the initial condition $X_0$ is independent of $\{ W_t \}$.
+
+### Moving Average Representation
+
+Iterating backwards from time $t$, we obtain
+
+$$
+X_t = a X_{t-1} + b +  c W_t
+        = a^2 X_{t-2} + a b + a c W_{t-1} + b + c W_t
+        = \cdots
+$$
+
+If we work all the way back to time zero, we get
+
+```{math}
+:label: ar1_ma
+
+X_t = a^t X_0 + b \sum_{j=0}^{t-1} a^j +
+        c \sum_{j=0}^{t-1} a^j  W_{t-j}
+```
+
+Equation {eq}`ar1_ma` shows that $X_t$ is a well defined random variable, the value of which depends on
+
+* the parameters,
+* the initial condition $X_0$ and
+* the shocks $W_1, \ldots W_t$ from time $t=1$ to the present.
+
+Throughout, the symbol $\psi_t$ will be used to refer to the
+density of this random variable $X_t$.
+
+### Distribution Dynamics
+
+One of the nice things about this model is that it's so easy to trace out the sequence of distributions $\{ \psi_t \}$ corresponding to the time
+series $\{ X_t\}$.
+
+To see this, we first note that $X_t$ is normally distributed for each $t$.
+
+This is immediate from {eq}`ar1_ma`, since linear combinations of independent
+normal random variables are normal.
+
+Given that $X_t$ is normally distributed, we will know the full distribution
+$\psi_t$ if we can pin down its first two moments.
+
+Let $\mu_t$ and $v_t$ denote the mean and variance
+of $X_t$ respectively.
+
+We can pin down these values from {eq}`ar1_ma` or we can use the following
+recursive expressions:
+
+```{math}
+:label: dyn_tm
+
+\mu_{t+1} = a \mu_t + b
+\quad \text{and} \quad
+v_{t+1} = a^2 v_t + c^2
+```
+
+These expressions are obtained from {eq}`can_ar1` by taking, respectively, the expectation and variance of both sides of the equality.
+
+In calculating the second expression, we are using the fact that $X_t$
+and $W_{t+1}$ are independent.
+
+(This follows from our assumptions and {eq}`ar1_ma`.)
+
+Given the dynamics in {eq}`ar1_ma` and initial conditions $\mu_0,
+v_0$, we obtain $\mu_t, v_t$ and hence
+
+$$
+\psi_t = N(\mu_t, v_t)
+$$
+
+The following code uses these facts to track the sequence of marginal
+distributions $\{ \psi_t \}$.
+
+The parameters are
+
+```{code-cell} julia
+a, b, c = 0.9, 0.1, 0.5
+
+mu, v = -3.0, 0.6  # initial conditions mu_0, v_0
+```
+
+Here's the sequence of distributions:
+
+```{code-cell} julia
+using StatsPlots
+
+sim_length = 10
+grid = range(-5, 7, length = 120)
+
+plt = plot()
+for t in 1:sim_length
+    mu = a * mu + b
+    v = a^2 * v + c^2
+    dist = Normal(mu, sqrt(v))
+    plot!(plt, grid, pdf.(dist, grid), label=L"\psi_{%$t}", linealpha=0.7)
+end
+plt
+```
+
+## Stationarity and Asymptotic Stability
+
+Notice that, in the figure above, the sequence $\{ \psi_t \}$ seems to be converging to a limiting distribution.
+
+This is even clearer if we project forward further into the future:
+
+```{code-cell} julia
+function plot_density_seq(;mu_0=-3.0, v_0=0.6, sim_length=60)
```

Comment:
With Julia you almost never want to have functions which are entirely using named arguments because it can't specialize the function to generate fast code.  In this case it probably doesn't matter, but I would typically want to do this with the `mu_0` and `v_0` as postiional args (defaults is fine if they are actually used,, as they are here) and leave the `sim_length` as a named arg?

With named tuple args for parameters, it is able to specialize basd on the type of the named tuple, which isn't a problem.  But that is overkill here as discussed.